mpas_analysis/shared/mpas_xarray/mpas_xarray.py

```diff
@@ -69,70 +152,216 @@ def subset_variables(ds, vlist):  # {{{
     # drop spurious coordinates
     ds = ds.drop(dropcoords)
 
-    assert len(ds.variables.keys()) > 0, \
-        'MPAS_XARRAY ERROR: Empty dataset is returned.\n' \
-        'Variables {}\nare not found within the dataset ' \
-        'variables: {}.'.format(vlist, allvars)
+    if len(ds.data_vars.keys()) == 0:
+        raise ValueError(
+                'Empty dataset is returned.\n'
+                'Variables {}\n'
+                'are not found within the dataset '
+                'variables: {}.'.format(variable_list, allvars))
+
+    return ds  # }}}
+
+
+def preprocess(ds, time_variable_name, variable_list, sel_values,
```

Comment:
I am trying to reorganize which functions are public and which are private.  This required moving several functions to the end of the file (since my understanding is that private functions should follow public ones whenever possible).  GitHub makes arbitrary choices about which functions it follows and which it loses track of in such cases.

In any case, the order of the functions was not arbitrary.  It is somewhat related to the history of me making a class and then undoing it per your previous requests.